From: Simon McVittie <smcv@collabora.com>
Date: Sun, 6 Sep 2020 17:32:26 +0100
Subject: tests: Make app chooser backend more debuggable

Instead of crashing out when our expectations are not met, return an
error that other components can log.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Applied-upstream: 1.7.3
---
 tests/backend/appchooser.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tests/backend/appchooser.c b/tests/backend/appchooser.c
index d8d3f7e..e7bf0f8 100644
--- a/tests/backend/appchooser.c
+++ b/tests/backend/appchooser.c
@@ -122,7 +122,13 @@ handle_choose_application (XdpImplAppChooser *object,
   g_assert_no_error (error);
 
   if (g_key_file_has_key (keyfile, "backend", "expect-no-call", NULL))
-    g_assert_not_reached ();
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             G_IO_ERROR,
+                                             G_IO_ERROR_FAILED,
+                                             "Did not expect ChooseApplication to be called here");
+      return TRUE;  /* handled */
+    }
 
   request = request_new (sender, arg_app_id, arg_handle);
 
